activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -45,7 +45,7 @@ def to_json(options = nil)
   end
 end
 
-[Object, Array, FalseClass, Float, Hash, Integer, NilClass, String, TrueClass, Enumerable].reverse_each do |klass|
+[Object, FalseClass, Float, Integer, NilClass, String, TrueClass, Enumerable, Array, Hash].reverse_each do |klass|
```

Comment:
This list is originally alphabetical order.

(`Object, (Array, FalseClass, Float, Hash, Integer, NilClass, String, TrueClass), Enumerable`)

Instead of moving `Array` and `Hash`, move `Enumerable` before the classes.

```suggestion
[Enumerable, Object, Array, FalseClass, Float, Hash, Integer, NilClass, String, TrueClass].reverse_each do |klass|
```